### DISCUSSION TITLE
Live sync of /home/username to onedrive folder /eos/home/username

### DISCUSSION THREAD

**Discussion by Mshmaitelly** [abraunegg/onedrive#2667] [Q&A] [answered]
Task:

Sync /home/username
to
Onedrive folder called eos_home_username
Myfiles/eos_home_username

I already installed onedrive linux client, the gui version and its tray app.
So far I am doing dry run uploads only, and I can see the log of the dry runs.
I set in the syn_list the eos_home_username

However, after the syc, as per the log, sync is excluding everything but a directory callled eos_home_username

What I want to do, is sync all the content (less hidden files) of the /home/username to a folder called eos_home_unsername in the Ondrive

How do I specify which Onedrive folder to sync to?

Thanks.

**Comment by abraunegg** [comment, marked as answer]
**IMPORTANT:** Do NOT sync your entire home directory to Microsoft OneDrive. This is 100% not advisable.

Your home directory contains many important files that should never be uploaded to a Cloud Service.

This being said, the solution you want to implement means changing all your directory structure, for your home - this is detailed in part here:  https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#upload-data-from-the-local-onedrive-folder-to-a-specific-location-on-onedrive

You will need to modify this to make this work for your home directory ... 

I would re-evaluate why you want to store your home directory on OneDrive in the first place - that is not a very good idea.

**Comment by Mshmaitelly** [comment]
I wanted to accomplish a kind of live backup for my home directory. More specifically, the data documents, user Audio/Video/Photos to be backed to Onedrive.  

Timeshift on my Arch Linux can incrementally back up my Linux system, however, as warned by timeshift, including the home directory content is not advised. Since, a restore would overwrite any new user data changes and addition between last snapshot and the time of restore.  I have pacman/yay hook, so anytime I do an update/upgrade to the packages, a fresh timeshift snapshot is made. 

I excluded the home directory from Timeshift's snapshot, and I was looking for a backup solution for my home directory.
So, I had in mind an upload-only (or full-sync) of the user data content of home directory to a specified Onedrive folder. 
What do you advise?

Thanks a lot for your response.

**Comment by abraunegg** [comment]
@Mshmaitelly 
> What do you advise?

Configure your `~/OneDrive` path following the advice previously linked to above. You then are storing all your local ~/OneDrive folder online under the online path of /Myfiles/eos_home_username/ ... that is the first step.

Then, in your home directory, symbolically link ~/OneDrive/Audio --> /actual/local/path/to/data/ ... repeat for all the folders you want to replicate online.

The net effect will be:

* Your local home directory data will be where you store it now (unchanged)
* This data will be backed up to Microsoft OneDrive online
* The data online will be stored where you want it to be online

**Reply by Mshmaitelly** [reply]
Fantastic. Thanks again.

**Comment by Mshmaitelly** [comment]
Update: It is working fantastically.
Timeshift is doing system back 
Onedrive is downing live backup of the user data dirs.
